lib/puppet/util/windows/user.rb

```diff
@@ -145,6 +145,111 @@ def load_profile(user, password)
   end
   module_function :load_profile
 
+  def get_rights(name)
+    user_info = Puppet::Util::Windows::SID.name_to_principal(name.sub(/^\.\\/, "#{Puppet::Util::Windows::ADSI.computer_name}\\"))
+    return "" if user_info.nil?
+
+    rights = []
+    rights_pointer = FFI::MemoryPointer.new(:pointer)
+    number_of_rights = FFI::MemoryPointer.new(:ulong)
+    policy_handle = new_lsa_policy_handle
+    sid_pointer = FFI::MemoryPointer.new(:byte, user_info.sid_bytes.length).write_array_of_uchar(user_info.sid_bytes)
+
+    result = LsaEnumerateAccountRights(policy_handle.read_pointer, sid_pointer, rights_pointer, number_of_rights)
+    check_lsa_nt_status_and_raise_failures(result, "LsaEnumerateAccountRights")
+
+    number_of_rights.read_ulong.times do |index|
+      right = LSA_UNICODE_STRING.new(rights_pointer.read_pointer + index * LSA_UNICODE_STRING.size)
+      rights << right[:Buffer].read_arbitrary_wide_string_up_to
+    end
+
+    result = LsaFreeMemory(rights_pointer.read_pointer)
+    check_lsa_nt_status_and_raise_failures(result, "LsaFreeMemory")
+
+    result = LsaClose(policy_handle.read_pointer)
+    check_lsa_nt_status_and_raise_failures(result, "LsaClose")
```

Comment:
Would it make sense to have the `new_lsa_policy_handle` yield and clean up after itself so we don't have to do it explicitly?

Something like:

```ruby
new_lsa_policy_handle do |policy_handle|
  # do things with the handle
end

def new_lsa_policy_handle
  # logic to create policy_handle
  yield policy_handle
  # clean up
  result = LsaClose(policy_handle.read_pointer)
  check_lsa_nt_status_and_raise_failures(result, "LsaClose")
end
```